examples/Model.elm

```diff
@@ -81,90 +111,136 @@ update msg model =
-                    ( { model | error = Just <| toString error }, Cmd.none )
+            ( { model | records = [], error = Nothing }
+            , fetchRecordList model
+            )
+
+        FetchRecordResponse (Ok record) ->
+            ( { model
+                | formData = recordToFormData record
+                , error = Nothing
+              }
+            , Cmd.none
+            )
+
+        FetchRecordResponse (Err error) ->
+            model |> updateError error
+
+        FetchRecordsResponse (Ok recordList) ->
+            ( { model
+                | records = recordList
+                , error = Nothing
+              }
+            , Cmd.none
+            )
+
+        FetchRecordsResponse (Err error) ->
+            model |> updateError error
+
+        CreateRecordResponse (Ok _) ->
+            ( { model | formData = initialFormData }
+            , fetchRecordList model
+            )
+
+        CreateRecordResponse (Err error) ->
+            model |> updateError error
 
         EditRecord recordId ->
             ( model, fetchRecord recordId )
 
-        EditRecordResponse response ->
-            case response of
-                Ok _ ->
-                    ( model, fetchRecordList )
+        EditRecordResponse (Ok _) ->
+            ( model, fetchRecordList model )
 
-                Err err ->
-                    ( { model | error = Just <| toString err }, Cmd.none )
+        EditRecordResponse (Err error) ->
+            model |> updateError error
 
         DeleteRecord recordId ->
             ( model, deleteRecord recordId )
 
-        DeleteRecordResponse response ->
-            case response of
-                Ok record ->
-                    ( { model
-                        | records = removeRecordFromList record model.records
-                        , error = Nothing
-                      }
-                    , fetchRecordList
-                    )
+        DeleteRecordResponse (Ok record) ->
+            ( { model
+                | records = removeRecordFromList record model.records
+                , error = Nothing
+              }
+            , fetchRecordList model
+            )
+
+        DeleteRecordResponse (Err error) ->
+            model |> updateError error
+
+        UpdateFormTitle title ->
+            let
+                formData =
+                    model.formData
+
+                updated =
+                    { formData | title = title }
+            in
+                ( { model
+                    | formData = updated
+                    , records = updateRecordInList updated model.records
+                  }
+                , Cmd.none
+                )
+
+        UpdateFormDescription description ->
+            let
+                formData =
+                    model.formData
+
+                updated =
+                    { formData | description = description }
+            in
+                ( { model
+                    | formData = updated
```

Comment:
These nested updates are kind of a bummer. In Haskell there's a thing called lenses, which are super confusing and complicated, but aim to solve this kind of problem. I guess in Elm there's http://package.elm-lang.org/packages/arturopala/elm-monocle/latest .